FirebaseSharedSwift/Sources/third_party/StructureEncoder/StructureEncoder.swift

```diff
@@ -233,6 +236,9 @@ public class StructureEncoder {
   /// The strategy to use for encoding keys. Defaults to `.useDefaultKeys`.
   open var keyEncodingStrategy: KeyEncodingStrategy = .useDefaultKeys
 
+  /// If `usePassthroughTypes` is set to `true`, then any value of a type conforming to StructureCodingPassthroughType will not be encoded, but left as is in the resulting structure
+  open var usePassthroughTypes: Bool = false
```

Comment:
Is there a reason why this would not be turned by default (or always)? The only issue I see is that some of the Firestore types might be treated as Passthrough types when used in RTDB and Functions. This however is already an error condition.

Reply:
Yes, exactly so that you could use the same models with Firestore, RTDB and callable functions:

```
struct Model: Codable {
   var coord: GeoPoint
}
```
Would encode as a native `GeoPoint` in Firestore, but use the perfectly valid `Codable` implementation of `GeoPoint` for RTDB and functions.